### PR TITLE
Feature/reputation system

### PR DESCRIPTION
**Pull request by SilliusSodus**
Reputation system is in place. Now the reputation will change always on leave for the classes and reputation for races are only changed on event. Right now a race event is always fired when someone leaves and a choice is automatically made until there is a better UI for it. I will not make the UI yet as I don't know yet when these events should fire and how the choice should be made.

### CODE REVIEW

File: behind-the-tankard/Assets/ScriptableObjects/AI/Reputation/Events/ReputationEventsStats.cs [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        public float adjustment;
+    }
+    public RepRaceAdjustmentEffects[] eventOptions;
+    //public Dictionary<Race,float> adjustmentsDict = new List<Dictionary<Race,float>>();
+
+    private void Start() {
+        //initDict();
+    }
+
+    /*private void initDict(){
```
Comment (YWAMWeessies) [inline behind-the-tankard/Assets/ScriptableObjects/AI/Reputation/Events/ReputationEventsStats.cs]: if this is not doing anything please remove it

File: behind-the-tankard/Assets/ScriptableObjects/AI/Reputation/ReputationStats.cs [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+using System.Collections.Generic;
+using UnityEngine;
+using System;
+
+[CreateAssetMenu(fileName = "ReputationStats", menuName = "ScriptableObjects/ReputationStats", order = 0)]
+public class ReputationStats : ScriptableObject
+{
+
+    public float generalAdjustment;
+    public Enum[] reps;
```
Comment (YWAMWeessies) [inline behind-the-tankard/Assets/ScriptableObjects/AI/Reputation/ReputationStats.cs]: enums start with caps

File: behind-the-tankard/Assets/ScriptableObjects/AI/Reputation/Events/ReputationEventsStats.cs (hunk 1 above)
Comment (SilliusSodus) [reply behind-the-tankard/Assets/ScriptableObjects/AI/Reputation/Events/ReputationEventsStats.cs]: Removed

File: behind-the-tankard/Assets/ScriptableObjects/AI/Reputation/ReputationStats.cs (hunk 2 above)
Comment (SilliusSodus) [reply behind-the-tankard/Assets/ScriptableObjects/AI/Reputation/ReputationStats.cs]: Wasn't used anymore, so I removed it.

Comment (TJ-Smits) [reply behind-the-tankard/Assets/ScriptableObjects/AI/Reputation/ReputationStats.cs]: Yes they do

File: behind-the-tankard/Assets/ScriptableObjects/AI/Reputation/ReputationStats.cs [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+    [System.Serializable]
+    public class RepAdjustOnLeave
+    {
+		public Mood mood;
+        public float adjustment;
+		public RepAdjustOnLeave(Mood m, float a){
```
Comment (TJ-Smits) [inline behind-the-tankard/Assets/ScriptableObjects/AI/Reputation/ReputationStats.cs]: Code alignment seems to be off?

Spaces and TAB are probably seen differently

File: behind-the-tankard/Assets/ScriptableObjects/AI/Reputation/ReputationStats.cs [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+            {
+                if (r.mood == m)
+                {
+                    repAdjustOnLeaveDict.Add(r.mood,r.adjustment);
+					exists = true;
+                    break;
+                }
+            }
+			if(!exists){
+				repAdjustOnLeaveDict.Add(m,generalAdjustment);
```
Comment (TJ-Smits) [inline behind-the-tankard/Assets/ScriptableObjects/AI/Reputation/ReputationStats.cs]: Code alignment here as well and slightly above this

File: behind-the-tankard/Assets/ScriptableObjects/AI/Reputation/ReputationStats.cs (hunk 3 above)
Comment (SilliusSodus) [reply behind-the-tankard/Assets/ScriptableObjects/AI/Reputation/ReputationStats.cs]: Yeah, it's really strange, but I don't see it happen in my editor. Can you look in your editor to see if it's also there?

File: behind-the-tankard/Assets/ScriptableObjects/AI/Reputation/ReputationStats.cs (hunk 4 above)
Comment (SilliusSodus) [reply behind-the-tankard/Assets/ScriptableObjects/AI/Reputation/ReputationStats.cs]: Same as above

File: behind-the-tankard/Assets/ScriptableObjects/AI/Reputation/Events/ReputationEventsStats.cs [hunk 5]
```diff
@@ -0,0 +1,22 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+using UnityEngine;
+
+[CreateAssetMenu(fileName = "ReputationEvent", menuName = "ScriptableObjects/ReputationEvents", order = 0)]
+public class ReputationEventStats : ScriptableObject {
```
Comment (MattijnP) [inline behind-the-tankard/Assets/ScriptableObjects/AI/Reputation/Events/ReputationEventsStats.cs]: Move all these brackets to a new line

File: behind-the-tankard/Assets/ScriptableObjects/AI/Reputation/ReputationStats.cs (hunk 3 above)
Comment (MattijnP) [reply behind-the-tankard/Assets/ScriptableObjects/AI/Reputation/ReputationStats.cs]: Looks like there is a mix of tabs and spaces.
This should be normalized to 4 spaces everywhere

File: behind-the-tankard/Assets/Scripts/Environment/reputation/ReputationSystem.cs [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+    private static Dictionary<Class,float> _classReps = new Dictionary<Class, float>();
+
+    private void Start()
+    {
+        //Needs to change when saving the game becomes a thing. Remember this.
+        initializeDicts();
+        stats.Init();
+    }
+
+    private static void initializeDicts()
```
Comment (MattijnP) [inline behind-the-tankard/Assets/Scripts/Environment/reputation/ReputationSystem.cs]: Why are all methods static?

Comment (MattijnP) [inline behind-the-tankard/Assets/Scripts/Environment/reputation/ReputationSystem.cs]: Rename all methods to start with a capital letter

Comment (SilliusSodus) [reply behind-the-tankard/Assets/Scripts/Environment/reputation/ReputationSystem.cs]: The tavern will always have one reputation so making it static ensures no incorrect variables get changed and allows every class to call the reputation with ease.

Comment (SilliusSodus) [reply behind-the-tankard/Assets/Scripts/Environment/reputation/ReputationSystem.cs]: One of these days I'll learn.. Done

File: behind-the-tankard/Assets/ScriptableObjects/AI/Reputation/ReputationStats.cs (hunk 3 above)
Comment (SilliusSodus) [reply behind-the-tankard/Assets/ScriptableObjects/AI/Reputation/ReputationStats.cs]: Looks fine in files changed now.

File: behind-the-tankard/Assets/ScriptableObjects/AI/Reputation/Events/ReputationEventsStats.cs (hunk 5 above)
Comment (SilliusSodus) [reply behind-the-tankard/Assets/ScriptableObjects/AI/Reputation/Events/ReputationEventsStats.cs]: Done.

File: behind-the-tankard/Assets/ScriptableObjects/AI/Reputation/ReputationStats.cs (hunk 4 above)
Comment (SilliusSodus) [reply behind-the-tankard/Assets/ScriptableObjects/AI/Reputation/ReputationStats.cs]: Done.

**Pull request merged**